docs/web/diff.md

```diff
@@ -0,0 +1,902 @@
+# Comparing analysis results (Diff)
+
+Compare the outstanding reports in a "before" (_baseline_) and an "after" (_new_ or newline) analysis sets, and display the differences in between them the in the same 
+format as `results`. A report is outstanding if all of the following is true:
+* its detection status is _new_, _reopened_ or _unresolved_,
```

Comment:
A report is outstanding in a given time T if all the following is true:
- if it was detected before T
- if it was closed after T, or not closed at all
- a report can be closed if becomes undetected, considered false positive, intentional bug.

The server stores the past states of a reports with certain simplifications (see section **), so it can tell for any past time points whether the report was outstanding or closed.

The local report directory only stores a single present state.

In CodeChecker the most recent status of a report is also represented in the detection status:
-_new_: the report is new compared to the last analysis
-_reopened_: the report is new compared to the last analysis and it is reappearing
-_unresolved_: the report is still detected
-_resolved_: the report is not detected anymore

This is useful to quickly see if for example the analysis set of the "master" branch is updated, what are the new/unresolved reports we still have to work with.

Reply:
I added the "outstanding report" definition when we diff tags/timestamps.

I added links to the detection status and review status docs.

For the rest, this is just not the place to describe what detection statuses are, but rather their own page. On the fact that we ignore detection statuses for tags/timestamps there already is paragraph.